### PR TITLE
libxl: dom0 fixes

### PR DESCRIPTION
**Pull request by jean-edouard**
OXT-435

Signed-off-by: Jed <lejosnej@ainfosec.com>

### CODE REVIEW

**Comment by jean-edouard** [comment]
Warning: this should be merged along with all the other libxl PRs, or it will break dom0.

**Comment by jean-edouard** [comment]
Here is the list of libxl PRs:
https://github.com/OpenXT/xenclient-oe/pull/463
https://github.com/OpenXT/xenclient-oe/pull/473
https://github.com/OpenXT/xenclient-oe/pull/474 (you are here)
https://github.com/OpenXT/xenclient-oe/pull/475
https://github.com/OpenXT/idl/pull/25
https://github.com/OpenXT/input/pull/17
https://github.com/OpenXT/manager/pull/43
https://github.com/OpenXT/network/pull/3
https://github.com/OpenXT/toolstack/pull/21
https://github.com/OpenXT/toolstack-data/pull/41
https://github.com/OpenXT/xsm-policy/pull/11

Built here: http://158.69.127.133:8020/builders/openxt-custom/builds/217

**Comment by jean-edouard** [comment]
Added https://github.com/OpenXT/installer/pull/27

File: recipes-core/base-files/base-files/fstab.xenclient-dom0 [hunk 1]
```diff
@@ -4,6 +4,7 @@ devpts                        /dev/pts      devpts mode=0620,gid=5
 tmpfs                         /var/volatile tmpfs  defaults,rootcontext=system_u:object_r:var_t:s0,size=2M      0  0
 tmpfs                         /var/cache    tmpfs  defaults,rootcontext=system_u:object_r:var_t:s0,size=100M    0  0
 tmpfs                         /var/lib/dbus tmpfs  defaults,rootcontext=system_u:object_r:system_dbusd_var_lib_t:s0,size=1M    0  0
+tmpfs                         /var/lib/xen  tmpfs  defaults,rootcontext=system_u:object_r:xend_var_lib_t:s0,size=1M    0  0
```
Comment (rossphilipson) [inline recipes-core/base-files/base-files/fstab.xenclient-dom0]: I assume this is another temp area xl needs. It seems we still have xend references all over the policy.

File: recipes-openxt/xenclient/xenclient-dom0-tweaks/xenstore-init [hunk 2]
```diff
@@ -2,3 +2,4 @@
 
 # add xenstore entries here
 
+xenstore-write domid 0
```
Comment (rossphilipson) [inline recipes-openxt/xenclient/xenclient-dom0-tweaks/xenstore-init]: I am not sure I understand what this is for?

Comment (rossphilipson) [inline recipes-openxt/xenclient/xenclient-dom0-tweaks/xenstore-init]: Yea it sounds like it is supposed to be there for all domains but it was missing for dom0. Ack.

File: recipes-core/base-files/base-files/fstab.xenclient-dom0 (hunk 1 above)
Comment (jean-edouard) [inline recipes-core/base-files/base-files/fstab.xenclient-dom0]: Yes and yes.

**Comment by rossphilipson** [comment]
Merging soon.

**Pull request merged**